actiontext/lib/action_text/attachable.rb

```diff
@@ -99,8 +99,12 @@ def previewable_attachable?
     end
 
     # Returns the attachable as JSON with the +attachable_sgid+ included.
-    def as_json(*)
-      super.merge("attachable_sgid" => persisted? ? attachable_sgid : nil)
+    def as_json(*args)
+      json = super
+      only = args[0]&.dig(:only)
+
+      json["attachable_sgid"] = persisted? ? attachable_sgid : nil if only.nil? || only && Array(only).map(&:to_s).include?("attachable_sgid")
+      json
     end
```

Comment:
Is this method overriding [ActiveModel::Serializers::JSON#as_json](https://edgeapi.rubyonrails.org/classes/ActiveModel/Serializers/JSON.html#method-i-as_json)? Is the result of `super` a `Hash` instance, which itself defines [Hash#as_json](https://github.com/rails/rails/blob/910eb470c5c2f1871117893fc33f5443bde8bf8d/activesupport/lib/active_support/core_ext/object/json.rb#L171-L192)?

If that's the case, would we be able to simplify this implementation to delegate to `super.as_json`?

```suggestion
    def as_json(options = nil)
      super.merge("attachable_sgid" => persisted? ? attachable_sgid : nil).as_json(options)
    end
```

Reply:
@seanpdoyle 

Thank you for your comment.
You're right, we can use [Hash#as_json](https://github.com/rails/rails/blob/910eb470c5c2f1871117893fc33f5443bde8bf8d/activesupport/lib/active_support/core_ext/object/json.rb#L171-L192)!
I fixed it, Could I take a look?

There is one trouble from the difference between [ActiveModel::Serializers::JSON#as_json](https://edgeapi.rubyonrails.org/classes/ActiveModel/Serializers/JSON.html#method-i-as_json) and [Hash#as_json](https://github.com/rails/rails/blob/910eb470c5c2f1871117893fc33f5443bde8bf8d/activesupport/lib/active_support/core_ext/object/json.rb#L171-L192) as below, 
so I had to manipulate args before I pass args to [Hash#as_json](https://github.com/rails/rails/blob/910eb470c5c2f1871117893fc33f5443bde8bf8d/activesupport/lib/active_support/core_ext/object/json.rb#L171-L192).

```ruby
# ActiveModel
# returns `id` string key with symbol`:id`
MyModel.new.as_json(only: :id)
#=> {"id" => 2}

# Hash
# does not return `id` string key with symbol`:id`
{"id" => 2}.as_json(only: :id)
#=> {}

{"id" => 2}.as_json(only: "id")
#=> {"id" => 2}
```